operators/src/processing/raster_type_conversion.rs

```diff
@@ -0,0 +1,237 @@
+use async_trait::async_trait;
+use futures::{stream::BoxStream, StreamExt, TryFutureExt, TryStreamExt};
+use geoengine_datatypes::{
+    primitives::{RasterQueryRectangle, SpatialPartition2D},
+    raster::{ConvertDataTypeParallel, DefaultNoDataValue, Pixel, RasterDataType, RasterTile2D},
+};
+use num_traits::AsPrimitive;
+use serde::{Deserialize, Serialize};
+
+use crate::engine::{
+    ExecutionContext, InitializedRasterOperator, Operator, QueryContext, QueryProcessor,
+    RasterOperator, RasterQueryProcessor, RasterResultDescriptor, SingleRasterSource,
+    TypedRasterQueryProcessor,
+};
+use crate::util::Result;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct RasterTypeConversionParams {
+    output_data_type: RasterDataType,
+}
+
+pub type RasterTypeConversionOperator = Operator<RasterTypeConversionParams, SingleRasterSource>;
+
+pub struct InitializedRasterTypeConversionOperator {
+    result_descriptor: RasterResultDescriptor,
+    source: Box<dyn InitializedRasterOperator>,
+}
+
+#[typetag::serde]
+#[async_trait]
+impl RasterOperator for RasterTypeConversionOperator {
+    async fn initialize(
```

Comment:
- should we ensure/enforce in and out datatype are different?
- do we want to throw errors if values are lost during conversion from larger to smaller datatype?

Reply:
* it should be a no-op. I think we can add that when specialization is available. 
* i thought of this but it will cause a lot of "not real" errors. Users should be aware that out of bounds will turn to nodata (or clamp later?)